### PR TITLE
Perps: modify funding payment calculation

### PR DESCRIPTION
**Pull request by witgaw**
* Modify TWAP calculations to exclude time spent in auctions

### CODE REVIEW

File: protocol/0053-PERP-product_builtin_perpetual_future.md
Comment (EVODelavega) [inline protocol/0053-PERP-product_builtin_perpetual_future.md:89]: Does this imply we should discard external data points if the market is still in opening auction, or simply skip the funding payment if the market is in opening auction? Do we still factor in the data for margin calculation?

Comment (witgaw) [reply protocol/0053-PERP-product_builtin_perpetual_future.md:89]: Let me know if the latest commit clarifies it.

File: protocol/0053-PERP-product_builtin_perpetual_future.md
Comment (davidsiska-vega) [inline protocol/0053-PERP-product_builtin_perpetual_future.md:94]: are we really deleting the calculation formula / code?

Comment (witgaw) [reply protocol/0053-PERP-product_builtin_perpetual_future.md:94]: That was the intention as it will be fairly convoluted to explain in pseduocode now and was hoping the description + detailed AC example would be enough, but can add it if you think otherwise.

Comment (witgaw) [reply protocol/0053-PERP-product_builtin_perpetual_future.md:94]: @davidsiska-vega @EVODelavega I've [reinstated the pseudocode](https://github.com/vegaprotocol/specs/pull/2001/commits/68155efe0faa89df76906bf558e9cb212cac4121). If you could give it a once over when you have a moment and approve if you're happy that'd be amazing :pray:

**Review by davidsiska-vega — DISMISSED** [review]

File: protocol/0053-PERP-product_builtin_perpetual_future.md
Comment (wwestgarth) [inline protocol/0053-PERP-product_builtin_perpetual_future.md:97]: So there is one thing that adds a layer of complication to all of this, and thats being able to support spot prices coming in out-of-order (which could happen in the general case where someone might be submitting this data a transactions by hand, and not eth-oracles). 

The current core code can handle this by working out where the new-old-point should be, unwinding the TWAP back, adding the new point and and winding forward again.

To be able to do this with these auction changes as well, it gets just that little bit more complicated. So, is there any possibility of just rejecting any new points that arrive which are before a point we already have, just to simplify it all? Or it is important to support it?

Comment (witgaw) [reply protocol/0053-PERP-product_builtin_perpetual_future.md:97]: @wwestgarth that's a very good point, we still want to maintain that property, so I've got an [absolute cop-out](https://github.com/vegaprotocol/specs/pull/2001/commits/486b5a987104ced08c44a1d14fbebad1f2aaabe1) of a solution as far as specs go:

> Note that depending on what type of oracle is used for the spot price it may be that the oracle points only become known shortly before or at the funding payment cue time, so the above pseudocode is just an illustration of how these quantities should be calculated and the implementation will need to be able to apply such calculation retrospectively.

What do you think? Just not sure if complicating this further in the spec itself adds much value. Happy to help figure it out in the actual implementation by pairing or taking on a `core` task focused on that if that'd help at all.

Comment (wwestgarth) [reply protocol/0053-PERP-product_builtin_perpetual_future.md:97]: The cop out is fine 😄 I've got an idea in my head for how to do it, but it felt fiddly enough to ask whether it was even in scope to support, and maybe save me some effort!

**Review by wwestgarth — APPROVED** [review]

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**
